component-api/src/main/java/org/talend/sdk/component/api/configuration/ui/widget/BaseOnSchema.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (C) 2006-2023 Talend Inc. - www.talend.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.talend.sdk.component.api.configuration.ui.widget;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import org.talend.sdk.component.api.configuration.ui.meta.Ui;
+import org.talend.sdk.component.api.meta.Documentation;
+
+@Ui
+@Documentation("Mark a table column is filled by component schema auto, only for studio")
+@Retention(RUNTIME)
+@Target({ PARAMETER, FIELD })
+public @interface BaseOnSchema {
```

Comment:
Thinking that Base**d**OnSchema naming is preferable.
```suggestion
public @interface BasedOnSchema {
```

Reply:
done